extra_data/file_access.py

```diff
@@ -213,19 +213,19 @@ def _evaluate_path_infos(self):
 
         if m:
             if m[1] == 'gpfs/exfel/exp':
-                self._path_infos = (m[5], m[2], m[3])  # ONC path.
+                self._path_infos = (m[5], m[2], m[3], int(m[4][1:]))  # ONC path.
             else:
-                self._path_infos = (m[2], m[3], m[4])  # Maxwell path.
+                self._path_infos = (m[2], m[3], m[4], int(m[5][1:]))  # Maxwell path.
         else:
-            self._path_infos = (None, None, None)
+            self._path_infos = (None, None, None, None)
 
     def _evaluate_filename_infos(self):
         m = self.euxfel_filename_pattern.match(osp.basename(self.filename))
 
         if m:
-            self._filename_infos = (m[1], m[2], int(m[3]))
+            self._filename_infos = (m[1], int(m[2][1:]), m[3], int(m[4]) if m[3] != 'OVERVIEW' else 0)
```

Comment:
```suggestion
            self._filename_infos = (m[1], int(m[2]), m[3], int(m[4]) if m[3] != 'OVERVIEW' else 0)
```

I think this was to slice off the `R` for run, but that's not in your capturing group, so it slices off the first digit instead, and will go wrong for run numbers over 999.

Reply:
Well caught!